src/main/java/seedu/address/model/tag/FreeTimeTag.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.tag;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Free Time Tag in Dormie.
+ */
+public class FreeTimeTag extends Tag {
+
+    public static final String MESSAGE_CONSTRAINTS = "Free Time Tag should be Mon-Sun:HHmm-HHmm (24hr format)";
+    public static final String VALIDATION_REGEX =
+            "^(Mon|Tue|Wed|Thu|Fri|Sat|Sun):(0[0-9]|1[0-9]|2[0-3])([0-5][0-9])-(0[0-9]|1[0-9]|2[0-3])([0-5][0-9])$";
```

Comment:
This is a good format to save as. Perhaps if we have extra time in future, we can try to further separate the day and time! :)